src/rpm/builder.rs

```diff
@@ -422,7 +417,6 @@ impl RPMBuilder {
     /// use an external signer to sing and build
     ///
     /// See `signature::Signing` for more details.
-    #[cfg(feature = "signature-meta")]
```

Comment:
This is where I disagree. `build_and_sign` is user visible API and should not be usable for skipping signatures. Imho `build` should also create the header, but not sign it.

Reply:
>This is where I disagree. `build_and_sign` is user visible API and should not be usable for skipping signatures.

What do you mean by "not be usable for skipping signatures"?  You have to pass an `S: signature::Signing<crate::signature::algorithm::RSA>` in order to use this API, so it can't be skipped.  For most users, that means it won't be very useful without enabling `signature-pgp`, but since we've discussed users writing their own implementation of this trait on other issues, that's not universally true.  

>Imho `build` should also create the header, but not sign it.

That's what it currently does (in this PR)?  It produces a signature header and populates it with digests, but not the signatures.

Reply:
On a separate note though, we could almost do away with `build_and_sign()` if we take `RPMPackage::sign()` and make it return `self` instead of `()`.

That way you could write this chain, which ought to be equivalent to `build_and_sign()`.

```
let pkg = RpmBuilder::new(....)
    [....]
    .build()
    .sign(signer);
```